e2e/pods_test.go

```diff
@@ -23,12 +26,30 @@ func TestPodLifecycle(t *testing.T) {
 	if !ok {
 		timeout = 300 * time.Second
 	}
-	cmd = kubectl("wait", "--for=condition=available", "--timeout="+timeout.String(), "deploy/nginx")
+	cmd = kubectl("wait", "--for=condition=ready", "--timeout="+timeout.String(), "pod/vk-e2e-hpa")
 	if out, err := cmd.CombinedOutput(); err != nil {
 		t.Fatal(string(out))
 	}
+	t.Log("success create pod")
+
+	// query metrics
+	deadline = time.Now().Add(5 * time.Minute)
+	for {
+		t.Log("query metrics ....")
+		cmd = kubectl("get", "--raw", "/apis/metrics.k8s.io/v1beta1/namespaces/vk/pods/vk-e2e-hpa")
```

Comment:
Instead of this use of deadline you could change `kubectl` to take a context that has a timeout set and we can bound that to the exec call (`exec.CommandContext` instead of `exec.Command`)

Reply:
yes, good to know the exec.CommandContext. But for this case, the command return failure or success immediately. I need to retry the command till it return the success. I think what I need is like [retry-go](https://github.com/avast/retry-go) library. But it not worth introduce a new package.

Reply:
Oh right, silly me this is for the loop not the individual command.